plugins/gateway/kong/plugin.go

```diff
@@ -17,48 +17,66 @@ package main
 
 import (
 	"fmt"
+	"os"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/Kong/go-pdk"
 	"github.com/Kong/go-pdk/server"
 	"github.com/go-openapi/strfmt"
 
-	"github.com/openclarity/apiclarity/plugins/api/client/client"
-	"github.com/openclarity/apiclarity/plugins/api/client/client/operations"
 	"github.com/openclarity/apiclarity/plugins/api/client/models"
 	"github.com/openclarity/apiclarity/plugins/common"
-	"github.com/openclarity/apiclarity/plugins/common/trace_sampling_client"
+	"github.com/openclarity/apiclarity/plugins/common/apiclarity_client"
+)
+
+const (
+	MaxBodySize = 1000 * 1000
 )
 
 var (
-	traceSamplingClient *trace_sampling_client.Client
-	telemetriesAPI      *client.APIClarityPluginsTelemetriesAPI
+	apiclarityClient *apiclarity_client.Client
+	discoveredApis   []string
+	lock             sync.RWMutex
 )
 
 type Config struct {
 	EnableTLS            bool   `json:"enable_tls"`
 	Host                 string `json:"host"`
-	TraceSamplingHost    string `json:"trace_sampling_host"`
 	TraceSamplingEnabled bool   `json:"trace_sampling_enabled"`
+	Token                string `json:"trace_source_token"`
 }
 
 func New() interface{} {
 	return &Config{}
 }
 
 func (conf Config) Access(kong *pdk.PDK) {
-	if conf.TraceSamplingEnabled && traceSamplingClient == nil {
-		_ = kong.Log.Info("Creating trace sampling client")
-		// TODO tls will not work since trace sampling manager is not supporting it currently
-		traceSampling, err := trace_sampling_client.Create(false, conf.TraceSamplingHost, common.SamplingInterval)
+	if conf.TraceSamplingEnabled && apiclarityClient == nil {
+		_ = kong.Log.Info(fmt.Sprintf("Host: '%v'", conf.Host))
+		_ = kong.Log.Info(fmt.Sprintf("TraceSamplingEnabled: '%v'", conf.TraceSamplingEnabled))
+		_ = kong.Log.Info(fmt.Sprintf("Token: '%v'", conf.Token))
+		_ = kong.Log.Info(fmt.Sprintf("SamplingInterval: '%v'", common.SamplingInterval))
+		discoveredApis = []string{}
```

Comment:
Maybe having a map could be faster, no need to iterate over the list for each trace

Reply:
I put these during implementation and keep it because it can help to see in log what is the config used to start the plugin. I can remove all of this if it disturb (it was not present before)

Reply:
No, I'm talking about `discoveredApis = []string{}` It's currently a list, and if I understand correctly, it's used to keep tracks of discovered apis, so, basically, it's just a set.
It was a suggestion to use a set instead of a list.
But at the end, the size of this list will never be very big, so iterrating on each element on each received traces won't take long.
As you prefer.